starters/cms/src/data.ts

```diff
@@ -0,0 +1,174 @@
+import type { CollectionItemData, ManagedCollection, ManagedCollectionField } from "framer-plugin"
+
+import { framer } from "framer-plugin"
+import { PLUGIN_KEYS } from "./constants"
+
+export type ManagedCollectionFieldType = ManagedCollectionField["type"]
+export type ManagedCollectionFields = readonly ManagedCollectionField[]
+export type FieldIds = ReadonlySet<ManagedCollectionField["id"]>
+
+type DataSourceField =
+    | {
+          type: ManagedCollectionFieldType
+      }
+    | {
+          type: "html"
+      }
+
+export interface DataSource {
+    id: string
+    fields: Record<string, DataSourceField>
+    items: Record<string, unknown>[]
+}
+
+export function getDataSourceIds(): readonly string[] {
+    return ["articles", "categories"]
+}
+
+export async function getDataSource(dataSourceId: string, abortSignal?: AbortSignal): Promise<DataSource> {
+    return await fetch(`/datasources/${dataSourceId}.json`, { signal: abortSignal }).then(response => response.json())
```

Comment:
I'm assuming this function is where I should add most of my logic? maybe worth a little comment on what that looks like?